docs/cyber.md

```diff
@@ -0,0 +1,80 @@
+1. [IdIndexer](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Estimator](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Estimator.html).
+   Given a dataframe, it creates an IdIndexerModel (described next) for categorical features which
+   contains the information to map each partition and column seen in the given dataframe to an id.
+   for each partition or one consecutive range for all partition and column values.
+2. [IdIndexerModel](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe maps each partition and column field to a consecutive integer id.
+   Partitions or column values not encountered in the estimator are mapped to 0.
+   The model can operate in two modes, either create consecutive integer id independently 
+3. [MultiIndexer](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Estimator](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Estimator.html).
+   Uses multiple IdIndexer to generate a MultiIndexerModel (described next) for categorical features which
+   contains multiple IdIndexers for multiple partitions and columns.
+4. [MultiIndexerModel](../src/main/python/mmlspark/cyber/ml/feature_engineering/indexers.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe maps each partition and column field to a consecutive integer id.
+   Partitions or column values not encountered in the estimator are mapped to 0.
+   The model can operate in two modes, either create consecutive integer id independently 
+
+# feature engineering: [scalers.py](../src/main/python/mmlspark/cyber/ml/feature_engineering/scalers.py)
+1. [StandardScalarScaler](../src/main/python/mmlspark/cyber/ml/feature_engineering/scalers.py)
+   is a SparkML [Estimator](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Estimator.html).
+   Given a dataframe it creates a StandardScalarScalerModel (described next) which normalizes
+   any given dataframe according to the mean and standard deviation calculated on the 
+   dataframe given to the estimator.
+2. [StandardScalarScalerModel](../src/main/python/mmlspark/cyber/ml/feature_engineering/scalers.py)
+   is a SparkML [Transformer](https://spark.apache.org/docs/2.2.0/api/java/index.html?org/apache/spark/ml/Transformer.html).
+   Given a dataframe with a value column x, the transformer changes its value as follows:
+   x'=(x-mean)/stddev, i.e., if the transformer is given the same dataframe the estimator 
+   was given then the value column will have a mean of 0.0 and a standard deviation of 1.0.
+3. [LinearScalarScaler](../src/main/python/mmlspark/cyber/ml/feature_engineering/scalers.py)
```

Comment:
Perhaps we should call this MinMaxScaler to dovetail with the sklearn equivalent 

https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MinMaxScaler.html

Reply:
Done